storage/innobase/xtrabackup/src/keyring_plugins.cc

```diff
@@ -158,8 +158,13 @@ dberr_t xb_set_encryption(fil_space_t *space) {
   return (fil_set_encryption(space->id, Encryption::AES, key, iv));
 }
 
-const char *TRANSITION_KEY_PRIFIX = "XBKey";
-const size_t TRANSITION_KEY_NAME_MAX_LEN = Encryption::SERVER_UUID_LEN + 2 + 45;
+const char *TRANSITION_KEY_PREFIX = "XBKey";
+const size_t TRANSITION_KEY_PREFIX_LEN =
+    sizeof(TRANSITION_KEY_PREFIX) / sizeof(char);
+const size_t TRANSITION_KEY_RANDOM_DATA_LEN = 32;
+const size_t TRANSITION_KEY_NAME_MAX_LEN = TRANSITION_KEY_PREFIX_LEN +
+                                           Encryption::SERVER_UUID_LEN +
+                                           TRANSITION_KEY_RANDOM_DATA_LEN + 1;
```

Comment:
Dont we need the length of v1 for reading and writing keys to v1? Should we instead create TRANSITION_KEY_NAME_MAX_LEN_V1 and _V2?

Reply:
This is not necessary, as I mentioned on https://github.com/percona/percona-xtrabackup/pull/1179#discussion_r715532010 this is just a stamp to make the key unique, all previous backups (with longer key) have the key saved at xtrabackup_keys file and preparing with a new version will be able to read that key and fetch vault to retrieve the secret necessary to un-encrypt tables. This is `TRANSITION_KEY_NAME_MAX_LEN` is only used on generating new transition key code path. Not used at prepare.

Reply:
Thanks for clarifying marce.

xb_tablespace_keys_load_one() creates a buffer of TRANSITION_KEY_NAME_MAX_LEN. This will be 13 chars less now. How do we deal with loading keys into the buffer of smaller size ?

Reply:
@satya-bodapati you are correct. We do need versioning on the transition key, otherwise, using PXB 8.0.27 we won't be able to read transition keys prior to this version (longer keys). I've adjusted that.